Test.Activities.Automation.ActivityLib/Models/Activity/Services/SyncActivityService.cs

```diff
@@ -0,0 +1,246 @@
+using Microsoft.SharePoint;
+using Test.Activities.Automation.ActivityLib.Models.Activity.Classes;
+using Test.Activities.Automation.ActivityLib.Models.Helpers;
+using Test.Activities.Automation.ActivityLib.Utils.Constants;
+
+namespace Test.Activities.Automation.ActivityLib.Models.Activity.Services.SyncActivityService
+{
+    public class SyncActivityService
+    {
+        private ILogger _logger;
+
+        public SyncActivityService(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        public void SyncActivities(IEnumerable<ActivityInfo> activities)
+        {
+            try
+            {
+                _logger?.LogInformation("Synchronizing activities");
+
+                if (activities == null)
+                {
+                    throw new Exception("Activities are null");
+                }
+
+                var now = DateTime.Now;
+                var minDate = new DateTime(now.Year, now.AddMonths(-1).Month, 1);
+                var maxDate = now.Date;
+
+                var ensureService = new EnsureService(_logger);
+
+                ensureService.CheckActivityDate(ref activities, minDate, maxDate);
+
+                using (var site = new SPSite(Constants.Host))
+                using (var web = site.OpenWeb(Constants.Web))
+                {
+                    var spActivities = GetSpActivities(web, minDate, maxDate);
+
+                    var members = GetSpMembers(web);
+
+                    ensureService.CheckActivityUser(ref activities, members);
+
+                    ensureService.CheckActivityPaths(ref activities, members);
+
+                    var itemsToUpdate = ensureService.Ensure(spActivities, activities, members);
+
+                    web.AllowUnsafeUpdates = true;
+
+                    UpdateSpActivities(itemsToUpdate, web);
+                }
+            }
+            catch (Exception e)
+            {
+                throw new Exception($"Synchronizing activities failed. {e.Message}");
+            }
+        }
+
+        private IEnumerable<Member> GetSpMembers(SPWeb web)
+        {
+            try
+            {
+                _logger?.LogInformation("Getting existing members from SP");
+
+                var spMentorsList = web.Lists.TryGetList(Constants.Lists.Mentors);
+                if (spMentorsList == null) throw new Exception("Getting SP mentor list failed");
+
+                var spRootMentorsList = web.Lists.TryGetList(Constants.Lists.RootMentors);
+                if (spRootMentorsList == null) throw new Exception("Getting SP root mentor list failed");
+
+                var members = new List<Member>();
+
+                foreach (var item in spMentorsList.GetItems().Cast<SPListItem>())
+                {
+                    var mentor = SPHelper.GetUserValue(item, Constants.Activity.Employee);
+
+                    var paths = SPHelper.GetMultiChoiceValue(item, Constants.Activity.Paths);
+
+                    members.Add(new Member
```

Comment:
You may specify `Member Member.FromMentor(SPListItem item)` and `Member Member.FromRootMentor(SPListItem item)` static methods and move common code to a separate method.